haystack/components/embedders/hugging_face_tei_text_embedder.py

```diff
@@ -135,8 +135,9 @@ def run(self, text: str):
 
         text_to_embed = self.prefix + text + self.suffix
 
-        embedding = self.client.feature_extraction(text=text_to_embed)
+        embedding = self.client.feature_extraction(text=[text_to_embed])
         # The client returns a numpy array
         embedding = embedding.tolist()
+        embedding = embedding[0]
```

Comment:
```suggestion
        embeddings = self.client.feature_extraction(text=[text_to_embed])
        # The client returns a numpy array
        embedding = embeddings.tolist()[0]
```

this should make things clearer